app/views/groups/show.html.erb

```diff
@@ -0,0 +1,24 @@
+<h1>Състезания за група <%= @group.name %></h1>
+<h2><%= @group.name %></h2> 
+<br>
+
+<% cache ["practice_area" || 1, Contest.last, Run.where(:user_id => current_user.id).last] do %>
```

Comment:
The cached key used here doesn't depend on `@group`. I am pretty sure that if you have it like this each group will show the same contests. You need to specify a unique cache key here, that depends on `@group` or the contests in @group. Something like:

```ruby
cache ["contest_group", @group, @group.contests.last, Run.where(:user_id => current_user.id).last]
```